eng/get-build-info.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+az config set extension.use_dynamic_install=yes_without_prompt
+
+function get_build_run () {
+    local pipeline_id="$1"
+    local pipeline_name="$2"
+    local azdo_org="$3"
+    local azdo_project="$4"
+    local tag="$5"
+
+    if [[ -n "$tag" ]]; then
+        build_runs=$(az pipelines runs list --organization "$azdo_org" --project "$azdo_project" --pipeline-ids "$pipeline_id" --tags "$tag")
+    else
+        build_runs=$(az pipelines runs list --organization "$azdo_org" --project "$azdo_project" --pipeline-ids "$pipeline_id")
+    fi
+
+    runs=$(echo "$build_runs" | jq -r '[.[] | { "result": .result, "id": .id, "sourceVersion": .sourceVersion }]')
+    run_count=$(echo "$runs" | jq 'length')
+
+    if [ "$run_count" != "1" ]; then
+        local tagged=''
+        if [[ -n "$tag" ]]; then
+            tagged=" tagged ${tag}"
+        fi
+        echo "##vso[task.logissue type=error]There are ${run_count} runs of ${pipeline_name}${tagged}. Please manually specify run ID to use."
+        echo "##vso[task.logissue type=error]Run IDs are: ${runs}"
+        exit 1
+    fi
+
+    run_id=$(echo "$runs" | jq -r '.[0].id')
+    run_source_version=$(echo "$runs" | jq -r '.[0].sourceVersion')
+
+    # run_result=$(echo "$runs" | jq -r '.[0].result')
+    # if [[ "$run_result" == "failed" ]]; then
+    #     echo "##vso[task.logissue type=error]: ${pipeline_name} run ID ${run_id} failed. Please manually specify a build ID to use instead. Exiting..."
+    #     exit 1
+    # fi
```

Comment:
**Question**: Should we have a parameter on the pipeline saying we allow failed runs? I am not sure we will get a green one for the VMR CI for preview 1

Reply:
Yes, I think we will need it for previews.